sample_app/app_utils.c

```diff
@@ -0,0 +1,714 @@
+      break;
+   case PNET_CONTROL_COMMAND_APP_RDY:
+      s = "APP_RDY";
+      break;
+   case PNET_CONTROL_COMMAND_RELEASE:
+      s = "RELEASE";
+      break;
+   default:
+      s = "<error>";
+      break;
+   }
+
+   return s;
+}
+
+const char * app_utils_event_to_string (pnet_event_values_t event)
+{
+   const char * s = "<error>";
+
+   switch (event)
+   {
+   case PNET_EVENT_ABORT:
+      s = "PNET_EVENT_ABORT";
+      break;
+   case PNET_EVENT_STARTUP:
+      s = "PNET_EVENT_STARTUP";
+      break;
+   case PNET_EVENT_PRMEND:
+      s = "PNET_EVENT_PRMEND";
+      break;
+   case PNET_EVENT_APPLRDY:
+      s = "PNET_EVENT_APPLRDY";
+      break;
+   case PNET_EVENT_DATA:
+      s = "PNET_EVENT_DATA";
+      break;
+   }
+
+   return s;
+}
+
+int app_utils_pnet_cfg_init_default (pnet_cfg_t * cfg)
+{
+   memset (cfg, 0, sizeof (pnet_cfg_t));
+
+   cfg->tick_us = APP_TICK_INTERVAL_US;
+
+   /* Identification & Maintenance */
+
+   cfg->im_0_data.im_vendor_id_hi = GET_HIGH_BYTE (APP_GSDML_VENDOR_ID);
+   cfg->im_0_data.im_vendor_id_lo = GET_LOW_BYTE (APP_GSDML_VENDOR_ID);
+
+   cfg->im_0_data.im_hardware_revision = 1;
+   cfg->im_0_data.im_sw_revision_prefix = APP_GSDML_SW_REV_PREFIX;
+   cfg->im_0_data.im_sw_revision_functional_enhancement = PNET_VERSION_MAJOR;
+   cfg->im_0_data.im_sw_revision_bug_fix = PNET_VERSION_MINOR;
+   cfg->im_0_data.im_sw_revision_internal_change = PNET_VERSION_PATCH;
+   cfg->im_0_data.im_revision_counter = 0; /* Only 0 allowed according
+                                                       to standard */
+   cfg->im_0_data.im_profile_id = APP_GSDML_PROFILE_ID;
+   cfg->im_0_data.im_profile_specific_type = APP_GSDML_PROFILE_SPEC_TYPE;
+   cfg->im_0_data.im_version_major = 1;
+   cfg->im_0_data.im_version_minor = 1;
+   cfg->im_0_data.im_supported = APP_GSDML_IM_SUPPORTED;
+   strcpy (cfg->im_0_data.im_order_id, APP_GSDML_ORDER_ID);
+   strcpy (cfg->im_0_data.im_serial_number, "00001");
+   strcpy (cfg->im_1_data.im_tag_function, "my function");
+   strcpy (cfg->im_1_data.im_tag_location, "my location");
+   strcpy (cfg->im_2_data.im_date, "2020-09-03 13:53");
+   strcpy (cfg->im_3_data.im_descriptor, "my descriptor");
+   strcpy (cfg->im_4_data.im_signature, ""); /* Functional safety */
+
+   /* Device configuration */
+   cfg->device_id.vendor_id_hi = GET_HIGH_BYTE (APP_GSDML_VENDOR_ID);
+   cfg->device_id.vendor_id_lo = GET_LOW_BYTE (APP_GSDML_VENDOR_ID);
+   cfg->device_id.device_id_hi = GET_HIGH_BYTE (APP_GSDML_DEVICE_ID);
+   cfg->device_id.device_id_lo = GET_LOW_BYTE (APP_GSDML_DEVICE_ID);
+
+   cfg->oem_device_id.vendor_id_hi = 0xc0;
+   cfg->oem_device_id.vendor_id_lo = 0xff;
+   cfg->oem_device_id.device_id_hi = 0xee;
+   cfg->oem_device_id.device_id_lo = 0x01;
+   strcpy (cfg->product_name, APP_GSDML_PRODUCT_NAME);
+
+   /* Timing */
+   cfg->min_device_interval = APP_GSDML_MIN_DEVICE_INTERVAL;
+
+   /* Network configuration */
+   cfg->send_hello = true;
+   cfg->num_physical_ports = 1;
```

Comment:
Is this later overwritten? If so, add a comment.

Reply:
OK
Updated docstring in header and added comment of this line